code/game/machinery/Sleeper.dm

```diff
@@ -90,6 +90,12 @@
 	close_machine(target)
 
 /obj/machinery/sleeper/attackby(obj/item/I, mob/user, params)
+	if(state_open)
+		to_chat(user, "<span class='warning'>[src] must be closed to [panel_open ? "close" : "open"] its maintenance hatch!</span>")
+		return
+	if(occupant)
+		to_chat(user, "<span class='warning'>[src] is currently occupied!</span>")
+		return
 	if(!state_open && !occupant)

```

Comment:
There is no point where this if will ever fail

Reply:
I'm not sure how to fix it, with `else return..()`?

Reply:
You already return if either state_open or occupant are truthy. By the time it gets here the code will always run inside this block so you may as well remove the if check.

Reply:
see https://github.com/tgstation/tgstation/blob/12ae904bf1006102e8c868c1236e2a7a650b2f42/code/game/machinery/harvester.dm#L133-L145

Reply:
Alright, that's doable